### PR TITLE
Remove let

### PR DESCRIPTION
**Pull request by nightpool**
remove let statements from extensions, because they've been causing problems.

### CODE REVIEW

File: Extensions/postarchive.js [hunk 1]
```diff
@@ -1346,11 +1346,11 @@ XKit.extensions.postarchive = {
 
 					}
 
-					for (let i = 0; i < XKit.extensions.postarchive.categories.length; i++) {
+					for (var ii = 0; j < XKit.extensions.postarchive.categories.length; ii++) {
```
Comment (nightpool) [inline Extensions/postarchive.js]: error

File: Extensions/blacklist.js [hunk 2]
```diff
@@ -1032,9 +1032,9 @@ XKit.extensions.blacklist = new Object({
 		if (XKit.extensions.blacklist.whitelisted.length <= 1) {
 			m_html = m_html + "<div class=\"xkit-blacklist-none\"><b>You have no whitelisted words.</b></div>";
 		} else {
-			for (let i = 0; i < XKit.extensions.blacklist.whitelisted.length; i++) {
-				if (XKit.extensions.blacklist.whitelisted[i] !== "") {
-					m_html = m_html + XKit.extensions.blacklist.create_blacklist_div(XKit.extensions.blacklist.whitelisted[i], true);
+			for (var j = 0; i < XKit.extensions.blacklist.whitelisted.length; j++) {
```
Comment (hobinjk) [inline Extensions/blacklist.js]: The condition still has i instead of j

**Comment by nightpool** [comment]
needs testing on the xkit_preferences changes

File: Extensions/xkit_preferences.js [hunk 3]
```diff
@@ -1590,263 +1590,266 @@ XKit.extensions.xkit_preferences = new Object({
 
 	},
 
-	return_extension_settings: function(extension_id) {
```
Comment (hobinjk) [inline Extensions/xkit_preferences.js]: I'm on mobile so I can't check this out right now, but what happened to the diff here?

**Comment by nightpool** [comment]
It's just very big

**Comment by blackjackkent** [comment]
Still testing, but there's definitely a bug in preferences -- only showing one customization option per extension.
there can only be one

**Comment by blackjackkent** [comment]
This LGTM now.

@nightpool i know you said in the chat that you are doing further testing yourself so 
@homu delegate=nightpool

**Comment by nightpool** [comment]
Did testing on this today

@homu r+

**Pull request merged**